### PR TITLE
Base off of third party VAT package we currently use

### PR DESCRIPTION
**Pull request by chrisrollins65**
https://digitalcrew.teamwork.com/app/tasks/21381120

Basically I just copy-pasted from the https://github.com/dannyvankooten/vat package we currently use to validate VATs and edited it to fit our own needs.

Having this under our control will allow us to more easily modify and maintain it as needed, since the external package seems almost abandoned.

### CODE REVIEW

File: numbers_test.go [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+		}
+
+		if test.valid != valid {
+			t.Errorf("Expected %v for %v, got %v", test.valid, test.number, valid)
+		}
+	}
+}
+
+func ExampleValidateNumber() {
+	vatNumber := "IT06823950966"
```
Comment (chrisrollins65) [inline numbers_test.go:177]: This was originally an Irish VAT number, but this test actually calls the VIES VAT service and it looks like right now VIES is reporting ALL Irish VATs as invalid (I tested with about 10 legit ones both via POST call and directly on their website).

So I changed it to an Italian one for now. Same in the `TestValidateNumberExistence` test below. But it worries me that Irish VATs do not work...

File: numbers_test.go [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+		}
+
+		if test.valid != valid {
+			t.Errorf("Expected %v for %v, got %v", test.valid, test.number, valid)
+		}
+	}
+}
+
+func ExampleValidateNumber() {
+	vatNumber := "BE0472429986"
```
Comment (chrisrollins65) [inline numbers_test.go:177]: This was originally an Irish VAT number, but this test actually calls the VIES VAT service and it looks like right now VIES is reporting ALL Irish VATs as invalid (I tested with about 10 legit ones both via POST call and directly on their website).

So I changed it to a Belgium one I found somewhere online for now. Same in the TestValidateNumberExistence test below. But it worries me that Irish VATs do not work...

Comment (chrisrollins65) [reply numbers_test.go:177]: Another question is whether we really should be calling the external VAT checker in these tests. If that server fails our tests break. It serves to help us keep informed if the service goes down (as it has helped me discover Ireland VATs aren't working right now), but maybe our tests should not depend on it...

Comment (chrisrollins65) [reply numbers_test.go:177]: Update - Ireland VATs are working again on VIES

File: numbers.go [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+	if bytes.Contains(xmlRes, []byte("INVALID_INPUT")) {
+		return nil, ErrInvalidVATNumber
+	}
+
+	// check if response contains "MS_UNAVAILABLE" string
+	if bytes.Contains(xmlRes, []byte("MS_UNAVAILABLE")) ||
+		bytes.Contains(xmlRes, []byte("MS_MAX_CONCURRENT_REQ")) {
+		return nil, ErrServiceUnavailable
+	}
+
```
Comment (chrisjtwomey) [inline numbers.go:136]: +1

File: numbers_test.go (hunk 2 above)
Comment (chrisjtwomey) [reply numbers_test.go:177]: Don't think the tests should make real calls to VIES. We should mock the responses, so we're not relying on the service actually working to test the code.

**Review by chrisjtwomey — APPROVED** [review]
Left one comment regarding tests. Everything else LGTM

File: numbers_test.go (hunk 2 above)
Comment (chrisrollins65) [reply numbers_test.go:177]: Yeah I agree, though it does provide a good quick way to test that VIES is working for all the different countries.

But I'll look into how we can go about mocking this and change these tests

**Comment by bakaat** [comment]
how's this going ?

**Comment by chrisrollins65** [comment]
> how's this going ?

Should be good to go. The tests are updated so they don't make VIES calls. Was waiting for Chris to look over my changes again but he's already approved this PR so I guess we should be good to merge even if he doesn't review my latest changes?

**Comment by bakaat** [comment]
nice one, I'd say so but lets poke @chrisjtwomey to have a final look there

File: numbers_test.go [hunk 4]
```diff
@@ -170,40 +176,108 @@ func TestValidateNumber(t *testing.T) {
...
+}
 
-	valid, _ = ValidateNumberExistence("NL123456789B01")
-	if valid {
-		t.Error("Expected NL123456789B01 to not be a valid VAT number.")
+func expectedViesResponse(vatNumber string, countryCode string, isValid bool) string {
+	isValidString := "true"
+	if !isValid {
+		isValidString = "false"
 	}
```
Comment (chrisjtwomey) [inline numbers_test.go]: Could use `strconv.FormatBool(isValid)` here?

**Comment by chrisjtwomey** [comment]
One comment but LGTM

File: numbers_test.go (hunk 4 above)
Comment (chrisrollins65) [reply numbers_test.go]: Cool, didn't know about that. Looks like it works

**Pull request merged**